internal/indexer/versionedscanner.go

```diff
@@ -15,6 +21,23 @@ type VersionedScanner interface {
 	Kind() string
 }
 
+// ConfigDeserializer can be thought of as a curried Unmarshal function.
```

Comment:
```suggestion
// ConfigDeserializer can be thought of as an Unmarshal function.
```
don't believe curried is the right word here as it's just a closure over another method.

Reply:
`Unmarshal` usually has the signature `func([]byte, interface{}) error`, so I was trying to point out it's that same concept with the `[]byte` closed over from a different scope.